File: 백준/Silver/11723. 집합/집합.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int m = Integer.parseInt(br.readLine());
        Set<Integer> s = new HashSet<>();
        StringBuilder sb = new StringBuilder();

        for (int i = 0; i < m; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            String cmd = st.nextToken();
            int n = 0;

            if (st.hasMoreTokens()){
                n = Integer.parseInt(st.nextToken());
            }

            switch (cmd) {
                case "add":
                    s.add(n);
                    break;
                case "remove":
                    s.remove(n);
                    break;
                case "check":
                    sb.append(s.contains(n) ? 1 : 0).append('\n');
                    break;
                case "toggle":
                    if (s.contains(n)) {
                        s.remove(n);
                    } else {
                        s.add(n);
                    }
                    break;
                case "all":
                    s.clear();
                    for (int j = 1; j <= 20; j++) {
                        s.add(j);
                    }
                    break;
                case "empty":
                    s.clear();
                    break;
            }
        }
        System.out.println(sb);
    }
}

File: 백준/Silver/11723. 집합/집합.py
import sys
input = sys.stdin.readline

m = int(input())
s = 0
for _ in range(m):
    cmd = input().split()
    if cmd[0] == 'add':
        s |= (1 << int(cmd[1]) - 1)
    elif cmd[0] == 'remove':
        s &= ~(1 << int(cmd[1]) - 1)
    elif cmd[0] == 'check':
        print(1 if s & (1 << int(cmd[1]) - 1) else 0)
    elif cmd[0] == 'toggle':
        s ^= (1 << int(cmd[1]) - 1)
    elif cmd[0] == 'all':
        s = (1 << 20) - 1
    elif cmd[0] == 'empty':
        s = 0